effect/default_effector.go

```diff
@@ -29,41 +29,46 @@ func NewDefaultEffector() *DefaultEffector {
 // MergeEffects merges all matching results collected by the enforcer into a single decision.
 func (e *DefaultEffector) MergeEffects(expr string, effects []Effect, results []float64) (bool, error) {
 	result := false
+	results[len(results)-1] = float64(-1)
 	if expr == "some(where (p_eft == allow))" {
 		result = false
-		for _, eft := range effects {
+		for i, eft := range effects {
 			if eft == Allow {
 				result = true
+				results[len(results)-1] = float64(i)
 				break
 			}
 		}
 	} else if expr == "!some(where (p_eft == deny))" {
 		result = true
-		for _, eft := range effects {
+		for i, eft := range effects {
 			if eft == Deny {
 				result = false
+				results[len(results)-1] = float64(i)
```

Comment:
Don't mess with the existing `results`. If you want to pass a variable, just add a new parameter

Reply:
Does adding a new parameter affect other implementations？

 what do you think of using `EffrctEx` ? We can add a function to determine what needs to be returned, I implemented it in the last commit

Reply:
`results` is unused.

Reply:
I think the point is don't add usage to `results`, which will make the code difficult to understand and less readable

Reply:
Maybe I think too much, I will listen to @hsluoyz

Reply:
Agreed @hsluoyz